cmd/rook/block/map.go

```diff
@@ -47,37 +46,34 @@ const (
 )
 
 var (
-	mountImageName     string
-	mountImagePoolName string
-	mountImagePath     string
+	mapImageName       string
+	mapImagePoolName   string
+	mapImagePath       string
+	mapFormatRequested bool
 )
 
-var mountCmd = &cobra.Command{
-	Use:   "mount",
-	Short: "Mounts a block image from the cluster as a local block device with the given file system path",
+var mapCmd = &cobra.Command{
+	Use:   "map",
+	Short: "Maps a block image from the cluster as a local block device and optionally formats and mounts it with the given file system path",
 }
 
 func init() {
-	mountCmd.Flags().StringVar(&mountImageName, "name", "", "Name of block image to mount (required)")
-	mountCmd.Flags().StringVar(&mountImagePoolName, "pool-name", "rbd", "Name of storage pool that contains block image to mount")
-	mountCmd.Flags().StringVar(&mountImagePath, "path", "", "File system path to mount block device on (required)")
+	mapCmd.Flags().StringVar(&mapImageName, "name", "", "Name of block image to map (required)")
+	mapCmd.Flags().StringVar(&mapImagePoolName, "pool-name", "rbd", "Name of storage pool that contains block image to map")
+	mapCmd.Flags().BoolVar(&mapFormatRequested, "format", false, "Format a filesystem after mapping")
+	mapCmd.Flags().StringVar(&mapImagePath, "mount", "", "Mount a filesystem on the indicated path")
 
-	mountCmd.MarkFlagRequired("name")
-	mountCmd.MarkFlagRequired("path")
+	mapCmd.MarkFlagRequired("name")
 
-	mountCmd.RunE = mountBlockEntry
+	mapCmd.RunE = mapBlockEntry
 }
 
-func mountBlockEntry(cmd *cobra.Command, args []string) error {
+func mapBlockEntry(cmd *cobra.Command, args []string) error {
 	rook.SetupLogging()
 
-	if err := flags.VerifyRequiredFlags(cmd, []string{"name", "path"}); err != nil {
-		return err
-	}
-
 	c := rook.NewRookNetworkRestClient()
```

Comment:
cobra strangely doesn't actually enforce required flags, thus we have to VerifyRequiredFlags ourselves